modules/message-lane/src/inbound_lane.rs

```diff
@@ -57,7 +87,13 @@ impl<S: InboundLaneStorage> InboundLane<S> {
 			return false;
 		}
 
+		// if there are more unconfirmed messages that we may accept, reject this message
+		if self.storage.max_unconfirmed_messages() == data.relayers.len() as MessageNonce {
```

Comment:
Yes, it was original idea from #339. There were multiple problems with this approach. Can't remember all, since a week has already passed, but here are some examples:
1) if you have some sort of queue (store relayers of last 1024 messages), it'll mean that every confirmation transaction after 1024 messages is guaranteed to have these additional >32kb bytes, unless you'll find a way to prove that reward has been paid (i.e. what is presented in this PR);
2) if you're going to introduce periods in blocks instead (store relayers + number of messages that they have relayed in last 1024 blocks), it'll require additional storage entry + separate transaction type to deliver that. Also - size of confirmation transaction may grow indefinitely (because we don't limit number of messages/1024 blocks);
3) since confirmation transaction requires source chain to know about target block' finality, it means that reward requires: (3.a) GRANDPA authorities to finalize target chain blocks in timely fashion (3.b) header relayers to relay target chain blocks and finality to source chain and (3.c) message relayer to submit delivery transaction. So there are too many other players (GRANDPA authorities, headers relayers, other relayers, nodes txpools) that may block claiming reward;
4) if you're going to introduce claim-reward transaction for relayers that have missed their opportunity (i.e. GRANDPA authorities have failed to finalize target chain blocks in timely fashion), it would mean that you'll need some way to mark reward as claimed (i.e. store unclaimed rewards separately OR change how pruning works).